drainer/util.go

```diff
@@ -191,3 +201,183 @@ func genDrainerID(listenAddr string) (string, error) {
 
 	return fmt.Sprintf("%s:%s", hostname, port), nil
 }
+
+// getParserFromSQLModeStr gets a parser and applies given sqlMode.
+func getParserFromSQLModeStr(sqlMode string) (*parser.Parser, error) {
+	mode, err := tmysql.GetSQLMode(sqlMode)
+	if err != nil {
+		return nil, err
+	}
+
+	parser2 := parser.New()
+	parser2.SetSQLMode(mode)
+	return parser2, nil
+}
+
+// collectDirFiles gets files in path.
+func collectDirFiles(path string) (map[string]struct{}, error) {
+	files := make(map[string]struct{})
+	err := filepath.Walk(path, func(_ string, f os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if f == nil {
+			return nil
+		}
+
+		if f.IsDir() {
+			return nil
+		}
+
+		name := strings.TrimSpace(f.Name())
+		files[name] = struct{}{}
+		return nil
+	})
+
+	return files, err
+}
+
+// getDBFromDumpFilename extracts db name from dump filename.
+func getDBFromDumpFilename(filename string) (db string, ok bool) {
+	if !strings.HasSuffix(filename, "-schema-create.sql") {
+		return "", false
+	}
+
+	idx := strings.LastIndex(filename, "-schema-create.sql")
+	return filename[:idx], true
+}
+
+// getTableFromDumpFilename extracts db and table name from dump filename.
+func getTableFromDumpFilename(filename string) (db, table string, ok bool) {
+	if !strings.HasSuffix(filename, "-schema.sql") {
+		return "", "", false
+	}
+
+	idx := strings.LastIndex(filename, "-schema.sql")
+	name := filename[:idx]
+	fields := strings.Split(name, ".")
```

Comment:
https://github.com/pingcap/dumpling/issues/131
Maybe we need to make sure there table name not contain special characters, or we have to parse it.